solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientBuilder.java

```diff
@@ -86,13 +87,27 @@ public B withFollowRedirects(boolean followRedirects) {
    * Solr servers.
    *
    * <p>For valid values see {@link org.apache.http.client.config.RequestConfig#getConnectTimeout()}
+   *
+   * @deprecated Please use {@link #withConnectionTimeout(long, TimeUnit)}
    */
+  @Deprecated(since = "9.2")
   public B withConnectionTimeout(int connectionTimeoutMillis) {
+    withConnectionTimeout(connectionTimeoutMillis, TimeUnit.MILLISECONDS);
+    return getThis();
+  }
+
+  /**
+   * Tells {@link Builder} that created clients should obey the following timeout when connecting to
+   * Solr servers.
+   *
+   * <p>For valid values see {@link org.apache.http.client.config.RequestConfig#getConnectTimeout()}
+   */
+  public B withConnectionTimeout(long connectionTimeout, TimeUnit unit) {
     if (connectionTimeoutMillis < 0) {
```

Comment:
you aren't referring to the parameter

Reply:
I suspect you had this problem because you renamed the parameter manually (as if your IDE was a simple text editor) to remove the "Millis" instead of telling your IDE (IntelliJ) to refactor-rename the parameter.  By far this is one of the most common IDE functions I do -- rename.  Memorize the hotkey.  I think recent IntelliJ versions even have some sort of after-the-fact detection of this to do the rename if you simply edit but it's not as reliable.